FileInfoLoader.cc

```diff
@@ -263,8 +261,11 @@ bool FileInfoLoader::FillHdf5ExtFileInfo(CARTA::FileInfoExtended* ext_info, std:
         // check dimensions
         casacore::uInt num_dims;
         casacore::IPosition data_shape;
+
+        auto counted_ptr = casacore::CountedPtr<casacore::HDF5File>(hdf_file_ptr);
```

Comment:
can you use a `CountedPtr` in the usual way? Wouldn't it be better to construct this straight away, instead of constructing `hdf_file_ptr`?

Reply:
It seems that you can. I also removed the intermediate variables in other places; they seemed unnecessarily verbose.